src/widget.rs

```diff
@@ -34,12 +44,41 @@ pub trait WidgetExtManual {
 
     fn override_symbolic_color(&self, name: &str, color: &RGBA);
 
+    fn connect_drag_data_get<F: Fn(&Self, &DragContext, &mut SelectionData, u32, u32) + 'static>(&self, f: F) -> u64;
+
     fn connect_map_event<F: Fn(&Self, &Event) -> Inhibit + 'static>(&self, f: F) -> u64;
 
     fn connect_unmap_event<F: Fn(&Self, &Event) -> Inhibit + 'static>(&self, f: F) -> u64;
 }
 
 impl<O: IsA<Widget> + IsA<Object>> WidgetExtManual for O {
+
+    fn drag_dest_set(&self, flags: DestDefaults, targets: &[TargetEntry], actions: DragAction) {
+        let stashes: Vec<_> = targets.iter().map(|e| e.to_glib_none()).collect();
+        let mut t = Vec::with_capacity(stashes.len());
+        for stash in stashes {
+            t.push(stash.0.clone());
+        }
+        unsafe { ffi::gtk_drag_dest_set(self.to_glib_none().0,
+                                        flags.to_glib(),
+                                        t.as_mut_ptr() as *mut ffi::GtkTargetEntry,
+                                        t.len() as i32,
+                                        actions.to_glib())};
+    }
+
+    fn drag_source_set(&self, start_button_mask: ModifierType, targets: &[TargetEntry], actions: DragAction) {
+        let stashes: Vec<_> = targets.iter().map(|e| e.to_glib_none()).collect();
+        let mut t = Vec::with_capacity(stashes.len());
+        for stash in stashes {
+            t.push(stash.0.clone());
+        }
+        unsafe { ffi::gtk_drag_source_set(self.to_glib_none().0,
+                                          start_button_mask.to_glib(),
+                                          t.as_mut_ptr() as *mut ffi::GtkTargetEntry,
```

Comment:
I did this because `gtk_drag_source_set` has a `mut` `targets` argument, but this actually is incorrect according to [the gtk docs](https://developer.gnome.org/gtk3/stable/gtk3-Drag-and-Drop.html#gtk-drag-source-set).